quantities/serialization/ScalarBuilder.cs

```diff
@@ -0,0 +1,91 @@
+            "metric" => GetMethod(nameof(CreateMetric), verification.Verify(metricUnits[unit]), prefix),
+            "imperial" => GetMethod(nameof(CreateImperial), verification.Verify(imperialUnits[unit]), prefix),
+            "any" => GetMethod(nameof(CreateNonStandard), verification.Verify(nonStandardUnits[unit]), prefix),
+            _ => throw new NotImplementedException($"{system} cannot be deserialized yet :-(")
+        };
+    }
+
+    private static Creator CreateSi<TSi>() where TSi : ISiUnit => i => i.Inject<Si<TSi>>();
+    private static Creator CreateSiAlias<TSi, TDim>() where TSi : ISiUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<Si<TSi>, Alias<TSi, TDim>>();
+    private static Creator CreateSi<TPrefix, TSi>() where TPrefix : IPrefix where TSi : ISiUnit => i => i.Inject<Si<TPrefix, TSi>>();
+    private static Creator CreateSiAlias<TPrefix, TSi, TDim>() where TPrefix : IPrefix where TSi : ISiUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<Si<TPrefix, TSi>, Alias<TPrefix, TSi, TDim>>();
+    private static Creator CreateMetric<TMetric>() where TMetric : IMetricUnit => i => i.Inject<Metric<TMetric>>();
+    private static Creator CreateMetricAlias<TMetric, TDim>() where TMetric : IMetricUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<Metric<TMetric>, Alias<TMetric, TDim>>();
+    private static Creator CreateMetric<TPrefix, TMetric>() where TPrefix : IPrefix where TMetric : IMetricUnit => i => i.Inject<Metric<TPrefix, TMetric>>();
+    private static Creator CreateMetricAlias<TPrefix, TMetric, TDim>() where TPrefix : IPrefix where TMetric : IMetricUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<Metric<TPrefix, TMetric>, Alias<TPrefix, TMetric, TDim>>();
+    private static Creator CreateImperial<TImperial>() where TImperial : IImperialUnit => i => i.Inject<Imperial<TImperial>>();
+    private static Creator CreateImperialAlias<TImperial, TDim>() where TImperial : IImperialUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<Imperial<TImperial>, Alias<TImperial, TDim>>();
+    private static Creator CreateNonStandard<TNonStandard>() where TNonStandard : INoSystemUnit => i => i.Inject<NonStandard<TNonStandard>>();
+    private static Creator CreateNonStandardAlias<TNonStandard, TDim>() where TNonStandard : INoSystemUnit, IInjectUnit<TDim> where TDim : Dimensions.IDimension => i => i.Inject<NonStandard<TNonStandard>, Alias<TNonStandard, TDim>>();
+
+    private static Creator GetMethod(String name, Type unit, Type? prefix = null)
+    {
+        var types = GetUnitTypeArgs(unit);
+        if (types.Count > 1) {
+            name = $"{name}Alias";
+        }
+        if (prefix != null) {
+            types.Insert(0, prefix);
+        }
+        var typeParameters = types.ToArray();
+        var genericMethod = GetGenericMethod(name, typeParameters.Length);
```

Comment:
I suppose you're doing this to avoid reflection. Correct me if I'm wrong.

Reply:
Here, I'm making full use of reflection. The place I'm avoiding reflections is in the `QuantityFactory` where I cache the instances of `IBuilder` returned by this class' public `Create` method.